ZenPacks/zenoss/OpenStackInfrastructure/api.py

```diff
@@ -47,8 +47,9 @@ def _runcommand(cmd):
             message = json.loads(stdout)['error']
         except Exception:
             message = stderr
-            
-        cmd[3] = '--api_key=***'
+
+        cmd = ['--api_key=***' if x.startswith('--api_key=') else x for x in cmd]
+
```

Comment:
Sorry to chime in late, but a more generic solution may be more robust. I've found several of these types of issues before: https://github.com/zenoss/ZenPacks.zenoss.DB2/blob/develop/ZenPacks/zenoss/DB2/lib/locallibs.py#L76-L83